android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt

```diff
@@ -159,6 +156,36 @@ class AddonManager(
         )
     }
 
+    /**
+     * Installs an [Addon] from the provided [url].
+     *
+     *  @param url the url pointing to either a resources path for locating the extension
+     *  within the APK file (e.g. resource://android/assets/extensions/my_web_ext) or to a
+     *  local (e.g. resource://android/assets/extensions/my_web_ext.xpi) or remote
+     *  (e.g. https://addons.mozilla.org/firefox/downloads/file/123/some_web_ext.xpi) XPI file.
+     * @param onSuccess (optional) callback invoked if the addon was installed successfully,
+     * providing access to the [Addon] object.
+     * @param onError (optional) callback invoked if there was an error installing the addon.
+     */
+    fun installAddon(
```

Comment:
Same here.

And I think the logic with `BLOCKED_PERMISSIONS` is no longer needed. Gecko verifies the privileged permissions too, and this would only work for installs from the Fenix manager.

Reply:
I think we should file a follow up bug to verify and confirm, the behaviour before removing the logic, as it looks like it's out side of the scope of the bug?

Reply:
Yes, file a follow-up to consider removing `BLOCKED_PERMISSIONS`.

Reply:
👍🏽

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1867297